### PR TITLE
Update slices dependency to use Compare

### PR DESCRIPTION
**Pull request by StephenButtolph**
Supersedes: #2405 and addresses: #2409

### CODE REVIEW

File: utils/sorting.go [hunk 1]
```diff
@@ -12,32 +12,23 @@ import (
...
-		return bytes.Compare(iHash, jHash) == -1
-	})
-}
-
-// Sorts a 2D byte slice.
-// Each byte slice is not sorted internally; the byte slices are sorted relative
-// to one another.
-func SortBytes[T ~[]byte](s []T) {
-	slices.SortFunc(s, func(i, j T) bool {
-		return bytes.Compare(i, j) == -1
```
Comment (StephenButtolph) [inline utils/sorting.go:40]: I felt like this function wasn't really needed anymore.

File: utils/sorting.go [hunk 2]
```diff
@@ -12,32 +12,23 @@ import (
...
 type Sortable[T any] interface {
-	Less(T) bool
+	Compare(T) int
 }
 
 // Sorts the elements of [s].
 func Sort[T Sortable[T]](s []T) {
-	slices.SortFunc(s, T.Less)
+	slices.SortFunc(s, T.Compare)
 }
```
Comment (StephenButtolph) [inline utils/sorting.go:24]: Honestly - this function could probably go as well

File: api/metrics/multi_gatherer.go [hunk 3]
```diff
@@ -91,7 +91,14 @@ func (g *multiGatherer) Register(namespace string, gatherer prometheus.Gatherer)
...
-		return *i.Name < *j.Name
+	slices.SortFunc(m, func(i, j *dto.MetricFamily) int {
+		switch {
+		case *i.Name < *j.Name:
+			return -1
+		case *i.Name > *j.Name:
+			return 1
+		default:
+			return 0
+		}
```
Comment (jmank88) [inline api/metrics/multi_gatherer.go:102]: There are std lib `Compare` and `Less` functions in `package cmp`: https://pkg.go.dev/cmp
```suggestion
		return cmp.Compare(*i.Name, *j.Name)
```

Comment (StephenButtolph) [reply api/metrics/multi_gatherer.go:102]: Hm, I don't think the `cmp` package is available for `go1.20`

Comment (StephenButtolph) [reply api/metrics/multi_gatherer.go]: I ended up backporting `Compare` for this change with a note to remove it after we update the golang version. Thanks for pointing that out

File: genesis/config_test.go [hunk 4]
```diff
@@ -11,56 +11,43 @@ import (
 	"github.com/ava-labs/avalanchego/ids"
 )
 
-func TestAllocationLess(t *testing.T) {
+func TestAllocationCompare(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply genesis/config_test.go:14]: The test body tests both `expected=-1` and `expected=1`

File: utils/sampler/weighted_array_test.go [hunk 5]
```diff
@@ -4,24 +4,39 @@
...
-func TestWeightedArrayElementLess(t *testing.T) {
-	require := require.New(t)
-
-	var elt1, elt2 weightedArrayElement
-	require.False(elt1.Less(elt2))
-	require.False(elt2.Less(elt1))
-
-	elt1 = weightedArrayElement{
-		cumulativeWeight: 1,
+func TestWeightedArrayElementCompare(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply utils/sampler/weighted_array_test.go:13]: The test body tests both `expected=-1` and `expected=1`

File: vms/avm/txs/operation.go [hunk 6]
```diff
@@ -48,16 +48,16 @@ type operationAndCodec struct {
 	codec codec.Manager
 }
 
-func (o *operationAndCodec) Less(other *operationAndCodec) bool {
+func (o *operationAndCodec) Compare(other *operationAndCodec) int {
 	oBytes, err := o.codec.Marshal(CodecVersion, o.op)
 	if err != nil {
-		return false
+		return 0
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/avm/txs/operation.go:54]: We don't have a logger available here... So it's a bit annoying to log this... But it should never happen

**Review by dhrubabasu — APPROVED** [review]

**Pull request merged**